app/celery/tasks.py

```diff
@@ -247,51 +241,21 @@ def send_sms(service_id, notification_id, encrypted_notification, created_at):
             job_id=notification.get('job', None),
             job_row_number=notification.get('row_number', None),
             status='sending',
-            created_at=datetime.strptime(created_at, DATETIME_FORMAT),
-            sent_at=sent_at,
-            sent_by=provider.get_name(),
-            content_char_count=template.replaced_content_count
+            created_at=datetime.strptime(created_at, DATETIME_FORMAT)
         )
-        statsd_client.timing_with_dates(
-            "notifications.tasks.send-sms.queued-for",
-            sent_at,
-            datetime.strptime(created_at, DATETIME_FORMAT)
-        )
-        dao_create_notification(notification_db_object, TEMPLATE_TYPE_SMS, provider.get_name())
-
-        try:
-            if service.research_mode:
-                send_sms_response.apply_async(
-                    (provider.get_name(), str(notification_id), notification['to']), queue='research-mode'
-                )
-            else:
-                provider.send_sms(
-                    to=validate_and_format_phone_number(notification['to']),
-                    content=template.replaced,
-                    reference=str(notification_id)
-                )
-
-                update_provider_stats(
-                    notification_id,
-                    'sms',
-                    provider.get_name()
-                )
+        dao_create_notification(notification_db_object, TEMPLATE_TYPE_SMS)
 
-        except SmsClientException as e:
-            current_app.logger.error(
-                "SMS notification {} failed".format(notification_id)
-            )
-            current_app.logger.exception(e)
-            notification_db_object.status = 'technical-failure'
-            dao_update_notification(notification_db_object)
+        send_sms_to_provider.apply_async((service_id, notification_id, encrypted_notification), queue='sms')
 
         current_app.logger.info(
             "SMS {} created at {} sent at {}".format(notification_id, created_at, sent_at)
         )
+
         statsd_client.incr("notifications.tasks.send-sms")
         statsd_client.timing("notifications.tasks.send-sms.task-time", monotonic() - task_start)
     except SQLAlchemyError as e:
         current_app.logger.exception(e)
+        raise self.retry(queue="retry", exc=e)
```

Comment:
We are retrying 5 times with 5 second intervals. Is that what we want when the db write fails?
Also the job is marked as complete but we will have missing notifications. Is that right?